atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/service/TransactionService.java

```diff
@@ -51,6 +51,20 @@ public interface TransactionService extends AutoCloseable, AsyncTransactionServi
     @Timed
     Map<Long, Long> get(Iterable<Long> startTimestamps);
 
+    /**
+     * This method MUST be called for each start timestamp before {@link #putUnlessExists(long, long) is ever called}.
+     * In practice, this means that a transaction must call this method before any information about its start timestamp
+     * is persisted into the KVS; i.e. KVS writes, writing to the sweep queue, etc.
+     *
+     * For safety, it is advisable that implementations of this method result in a no-op if this method is called after
+     * a successful call of {@link #putUnlessExists(long, long)} for the same startTimestamp.
```

Comment:
I see. Do we know when this might happen? I was thinking that before this call is made there isn't any information about the transaction's start timestamp in the KVS anyway, so it's not really feasible for others to have aborted me.

Reply:
Think this maybe have more as a protection against bad usage of this api.